### PR TITLE
AAG Stats: Format the dates of the stats labels.

### PR DESCRIPTION
**Pull request by dereksmart**
Fixes: #3873 

To test: 
- Activate stats and view the tooltips and x-axis labels for days, weeks, and months.

### CODE REVIEW

File: _inc/client/at-a-glance/stats.jsx [hunk 1]
```diff
@@ -150,7 +162,7 @@ const DashStatsBottom = React.createClass( {
 				<div className="jp-at-a-glance__stats-summary-bestday">
 					<p className="jp-at-a-glance__stat-details">Best overall day</p>
```
Comment (beaulebens) [inline _inc/client/at-a-glance/stats.jsx]: i18n

Comment (beaulebens) [inline _inc/client/at-a-glance/stats.jsx]: Unless you're handling that in a different full-sweep -- noticed there are others in this file as well.

File: _inc/client/at-a-glance/stats.jsx [hunk 2]
```diff
@@ -30,6 +31,16 @@ const DashStats = React.createClass( {
 		forEach( window.Initial_State.statsData[unit].data, function( v ) {
 			let label = v[0];
 			let views = v[1];
+
+			if ( 'day' === unit ) {
+				label = moment( v[0] ).format( 'MMM D' );
+			} else if ( 'week' === unit ) {
+				label = label.replace( /W/g, '-' );
+				label = __( 'Week of ' ) + moment( label ).format( 'MMM D' );
```
Comment (eliorivero) [inline _inc/client/at-a-glance/stats.jsx]: @dsmart maybe this should use a placeholder for better translation?

**Comment by beaulebens** [comment]
Looks great. The only one where there's any ambiguity is on the YEARS view, since it spans multiple years -- is there space to include the year in addition to the month? Or does it all start running into each other?

File: _inc/client/at-a-glance/stats.jsx (hunk 1 above)
Comment (dereksmart) [inline _inc/client/at-a-glance/stats.jsx]: It's been done in another PR

**Comment by dereksmart** [comment]
We currently don't show years in Jetpack stats either.  This was meant to mimic that functionality and "years" falls in the "enhanced stats" bucket.

**Comment by dereksmart** [comment]
@eliorivero @beaulebens I've updated this

**Comment by ebinnion** [comment]
LGTM.
